tests/atoms/gaia_apps.js

```diff
@@ -181,20 +181,31 @@ var GaiaApps = {
     var apps = window.wrappedJSObject.applications ||
                 window.wrappedJSObject.Applications;
     var app = apps.getByManifestURL(manifestURL);
-    var appName, launchPath;
-
-    if (entryPoint) {
-      if (app.manifest.entry_points[entryPoint]) {
-        appName = app.manifest.entry_points[entryPoint].name;
-        launchPath = app.manifest.entry_points[entryPoint].launch_path;
-      } else {
-        app = null;
-      }
-    } else {
-      appName = app.manifest.name;
-      launchPath = app.manifest.launch_path;
+    var appManifestURL, launchPath;
+
+    if(app != null){
+      app.onsuccess = function(){
+        if (entryPoint) {
+          if (app.manifest.entry_points[entryPoint]) {
+            appManifestURL = app.manifest;
+            launchPath = app.manifest.entry_points[entryPoint].launch_path;
+          } else {
+            app = null;
+          }
+        } else {
+          appManifestURL = app.manifest;
+          launchPath = app.manifest.launch_path;
+        }
+      GaiaApps.sendLocateResponse(callback, app, appManifestURL, 
+        launchPath, entryPoint);
+      marionetteScriptFinished(true);
```

Comment:
This line is not needed here, we already do the callback action in `sendLocateResponse()`